src/wasi/stat.rs

```diff
@@ -0,0 +1,60 @@
+use anyhow::{anyhow, bail, Result};
+use javy_plugin_api::javy::{
+    quickjs::{Ctx, Object as JObject, Value},
+    Args,
+};
+
+use super::{preview_1, process_error, FileType, Filestat};
+
+/// Get the file status of a file at the given path.
+/// This function is used to get the file status of a file at the given path.
+/// - `dirfd`: The directory file descriptor of the file.
+/// - `lookup_flags`: Flags for looking up the file descriptor.
+/// - `path`: The path of the file to get the status of.
+pub fn wasi_preview1_path_filestat_get(args: Args<'_>) -> Result<Value<'_>> {
+    let (cx, args) = args.release();
+    let args_pat: &[Value<'_>] = &args.0;
+    let [dirfd, lookup_flags, path, ..] = args_pat else {
+        bail!(
+            "wasi_preview1_path_filestat_get expects 3 parameters: the dirfd and path, Got: {} parameters.",
+            args.len()
+        );
+    };
+    let dirfd = dirfd
+        .as_int()
+        .ok_or_else(|| anyhow!("dirfd must be a number"))?;
+    let lookup_flags = lookup_flags
+        .as_int()
+        .ok_or_else(|| anyhow!("lookup_flags must be a number"))?;
+    let path = path
+        .as_string()
+        .ok_or_else(|| anyhow!("path must be a string"))?
+        .to_string()
+        .map_err(|_| anyhow!("invalid UTF-8 in path"))?;
```

Comment:
should 3 params be strictly enforced?
```suggestion
    if args.len() != 3 {
        bail!(
            "wasi_preview1_path_filestat_get expects exactly 3 parameters \
             (dirfd, lookup_flags, path); got {}.",
            args.len()
        );
    }
    let dirfd        = &args_pat[0];
    let lookup_flags = &args_pat[1];
    let path         = &args_pat[2];
```